cli/src/movement_util.rs

```diff
@@ -201,3 +220,34 @@ pub(crate) fn move_to_commit(
     tx.finish(ui, format!("{cmd}: {current_short} -> {target_short}"))?;
     Ok(())
 }
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, serde::Deserialize)]
+#[serde(rename_all(deserialize = "kebab-case"))]
+pub enum MovementEditMode {
+    #[default]
+    Auto,
+    Always,
+    Never,
+}
+
+#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, serde::Deserialize)]
+#[serde(rename_all(deserialize = "kebab-case"))]
+pub struct MovementSettings {
+    edit: MovementEditMode,
+}
+
+impl MovementSettings {
+    fn edit_mode(&self) -> MovementEditMode {
+        self.edit
+    }
+}
+
+impl TryFrom<&Config> for MovementSettings {
+    type Error = CommandError;
+
+    fn try_from(config: &Config) -> Result<Self, CommandError> {
+        config
+            .get::<Self>("ui.movement")
+            .map_err(|err| config_error_with_message("Invalid `ui.movement`", err))
```

Comment:
nit: you can inline this to caller. If we drop support for `auto`, this will be just `config.get_bool("ui.movement")?`.

Reply:
Done in #4302